src/Sarif/Readers/DeferredList.cs

```diff
@@ -109,6 +111,22 @@ private void BuildPositions(JsonPositionedTextReader reader, long currentOffset)
             _count = positions.Count;
         }
 
+        /// <summary>
+        ///  Add a just-in-time method to transform Results after deserialization but before getter returns them.
+        /// </summary>
+        /// <param name="transformer">Method to change Result instance and return updated Result</param>
+        public void AddTransformer(Func<T, T> transformer)
+        {
+            if (_transformer == null)
+            {
+                _transformer = transformer;
+            }
+            else
+            {
+                _transformer = (item) => transformer(_transformer(item));
```

Comment:
>transformer [](start = 41, length = 11)

This code is non-obvious in terms of what gets evaluated when. Can you clarify in a code comment? May be faster to discuss this in a call.

Reply:
Yes, comment added.